### DISCUSSION TITLE
selectAtom default value

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [pmndrs/jotai#1486] [Q&A] [answered]
See how I modified the [selectAtom](https://jotai.org/docs/utils/select-atom#examples) example by providing a default value.

Is this usage recommended?  If not, what would be the recommended way to provide a default value when _person.name_ is not defined?

```
const nameAtom = selectAtom(personAtom, (person) => person.name)
const nameAtom = selectAtom(personAtom, (person) => person.name || { first: 'John', last: 'Smith' });
```

**Comment by dai-shi** [comment, marked as answer]
Using a default value is fine, but you want to make it stable to avoid infinite loop.
```js
const defaultName = { first: 'John', last: 'Smith' };
const nameAtom = selectAtom(personAtom, (person) => person.name || defaultName);
```
